tests/unit/test_oidc.py

```diff
@@ -41,12 +47,20 @@ class DummyConfig:
         sasl_oauthbearer_expected_issuer="https://oidcprovider.com",
         sasl_oauthbearer_expected_audience="accounts-audience",
         sasl_oauthbearer_sub_claim_name="sub",
+        sasl_oauthbearer_authorization_enabled=False,
```

Comment:
I guess we still need to add unit tests for these. 
Also the `get_roles_from_claim_path()` and `authorize_request()`.

Integration tests would be nice, but can be a follow up.